torch/csrc/utils/pybind.h

```diff
@@ -57,6 +59,39 @@ template<> struct type_caster<torch::autograd::Variable> {
   }
 };
 
+template<> struct type_caster<at::IntList> {
```

Comment:
Is the `type_caster` struct guaranteed to stay live for the duration of the function call?

Reply:
I must admit I don't  have an authoritative documentation quote, but my impression is that they are.
The caster for `char *` seems to be holding on to a caster to `std::string` and then passing out its `c_str()`:
https://github.com/pybind/pybind11/blob/77374a7e5f48b64ee8b51de0a7d3783177a67716/include/pybind11/cast.h#L1293

Reply:
That convinces me!